### PR TITLE
Allow building Flann/libsvm to be optional

### PR DESCRIPTION
**Pull request by danlamanna**

### CODE REVIEW

**Comment by danlamanna** [comment]
I suppose (depending on the outcome of #284) libsvm building should be optional as well?

File: TPL/CMakeLists.txt [hunk 1]
```diff
@@ -40,26 +41,28 @@ ExternalProject_Add(libSVM
...
-  CMAKE_ARGS
+if (SMQTK_BUILD_FLANN)
+  set(FLANN_DIR "${CMAKE_CURRENT_LIST_DIR}/flann-1.8.4")
+  ExternalProject_Add(flann
+    PREFIX            "${TPL_BUILD_PREFIX}"
+    SOURCE_DIR        "${FLANN_DIR}"
+    BUILD_IN_SOURCE   1
+    CMAKE_GENERATOR   ${gen}
+    CMAKE_ARGS
     -DBUILD_MATLAB_BINDINGS:BOOL=OFF
```
Comment (Purg) [inline TPL/CMakeLists.txt]: Indent the CMake arguments over with the rest of the code block.

**Comment by danlamanna** [comment]
Should I amend this to include doing the same for libsvm, or open a different pr, or not do that altogether?

**Comment by Purg** [comment]
You can add the option for libSVM building here, too, just change the PR name to make it inclusive of that change as well.

**Comment by danlamanna** [comment]
@Purg PTAL

File: docs/release_notes/v0.7.0.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ Build System
   * Added ``setup.py`` script in support of installation by ``pip``. Updated
     CMake code to install python components via this scripts.
 
+  * Added ``SMQTK_BUILD_FLANN`` to CMake for optionally building Flann (defaults ON).
```
Comment (Purg) [inline docs/release_notes/v0.7.0.md]: Adjust or add note mentioning libSVM, too.

**Pull request merged**